Fixes #13542 - Add in the translation scaffolding for foreman-tasks

bkearney opened the pull request:
After this is merged. It will need to be added to transifex @claer, @stbenjam, @orrabin

domcleal commented on locale/action_names.rb:
```diff
@@ -0,0 +1,7 @@
+# Autogenerated!
+_("Delete architecture")
+_("Import puppetclasses")
```
"Puppet classes"

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:89
+msgid "Resume specific tasks by id"
```
`ID`

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:128
+msgid "Sort field and order, eg. 'name DESC'"
```
`e.g.`

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:167
+msgid "The id of the step inside the execution plan to send the event to "
```
`ID`, trailing whitespace

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:216
+msgid "Search_Params %s not supported"
```
Strange capitalisation of the first bit?

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/tasks_controller.rb:34
+msgid "The task is not cancellable at the moment."
```
cannot be cancelled?

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:28
+#: ../app/views/foreman_tasks/recurring_logics/index.html.erb:22
+#: ../app/views/foreman_tasks/task_groups/recurring_logic_task_groups/_recurring_logic_task_group.html.erb:17
+msgid "-"
```
Needs more context or to be removed.

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:90
+msgid "is minute (range: 0-59)"
```
A little more context may help, even a TRANSLATORS comment.

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:98
+msgid "\"Cron line format 'a b c d e', where:<br><ol type=\\\"a\\\">#{options}</ol>\""
```
Leading/trailing quotes, HTML and interpolation should be removed.

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/lib/actions/action_with_sub_plans.rb:15
+msgid "%{total} tasks, %{success} success, %{failed} fail"
```
missing plural forms?

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:136
+msgid "On"
```
Context may help.

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:131
+msgid "At"
```
Missing context?

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/lib/actions/foreman/puppetclass/import.rb:18
+msgid "Import puppetclasses"
```
"Puppet classes"

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/lib/actions/proxy_action.rb:60
+msgid "\"The smart proxy task '#{output[:proxy_task_id]}' failed.\""
```
Quotes and interpolation need to be removed.

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+msgstr ""
+
+#: ../app/models/foreman_tasks/triggering.rb:99
+msgid "\"#{cronline} is not valid format of cronline\""
```
Quotes and interpolation need to be removed.  "cronline" might need a space.

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+
+#: ../app/views/foreman_tasks/recurring_logics/index.html.erb:6
+#: ../app/views/foreman_tasks/task_groups/recurring_logic_task_groups/_recurring_logic_task_group.html.erb:8
+msgid "Cronline"
```
ditto above, may need a space?

domcleal commented on locale/foreman_tasks.pot:
```diff
@@ -0,0 +1,725 @@
+
+#:
+#: ../app/views/foreman_tasks/tasks/dashboard/_latest_tasks_in_error_warning.html.erb:1
+msgid "Latest Warning/Error Tasks "
```
Trailing whitespace.

bkearney commented:
@domcleal I addressed all your comments except 1 (see below). I did a force push with the latest strings.   The one comment I did not addres was

> +
> +#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:98
> +msgid "\"Cron line format 'a b c d e', where:<br><ol type=\\\"a\\\">#{options}</ol>\""

Leading/trailing quotes, HTML and interpolation should be removed.

I removed the interpolation, but it seems that using the popover command you need to pass in html snippets.

domcleal commented on app/helpers/foreman_tasks/foreman_tasks_helper.rb:
```diff
@@ -87,6 +87,7 @@ def recurring_mode_fieldset(f, triggering)
 
     def cronline_fieldset(f, triggering)
       options = [
+        # TRANSLATORS: these translations are ferring to an option which is a time interval
```
referring, and this comment won't work across multiple lines I don't think?

domcleal commented on app/helpers/foreman_tasks/foreman_tasks_helper.rb:
```diff
@@ -95,7 +96,7 @@ def cronline_fieldset(f, triggering)
       help = content_tag(:span, nil, :class => 'help-inline') do
         popover(_('Explanation'),
-                _("Cron line format 'a b c d e', where:<br><ol type=\"a\">#{options}</ol>"))
+                _("Cron line format 'a b c d e', where:<br><ol type=\"a\">%s</ol>") % (options))
```
Pass the HTML in through interpolation rather than in the source string.

bkearney commented:
@domcleal addressed both comments. I did need to replicate the TRANSLATORS comment because it did not get applied to subsequent translations.

domcleal commented on locale/action_names.rb:
```diff
@@ -0,0 +1,54 @@
+# Autogenerated!
+_("Update package")
```
These look like they're from a plugin you have loaded using foreman-tasks, not foreman-tasks itself.  Wrong project?

bkearney commented:
@domclel updated. I was not aware that string extraction required that only a single plugin be active.

domcleal commented:
It doesn't usually, it's probably the gettext extensions within foreman-tasks.

The POT looks OK for translation to me now.

iNecas commented:
Thanks @bkearney and @domcleal
